frontend/packages/dev-console/src/components/pipelines/pipeline-builder/PipelineBuilderForm.tsx

```diff
@@ -53,101 +49,118 @@ const PipelineBuilderForm: React.FC<PipelineBuilderFormProps> = (props) => {
-          <PipelineParameters addLabel="Add Parameters" fieldName="params" />
-        </div>
+          <div>
+            <h2>Parameters</h2>
+            <PipelineParameters addLabel="Add Parameters" fieldName="params" />
+          </div>
 
-        <div>
-          <h2>Resources</h2>
-          <PipelineResources addLabel="Add Resources" fieldName="resources" />
-        </div>
+          <div>
+            <h2>Resources</h2>
+            <PipelineResources addLabel="Add Resources" fieldName="resources" />
+          </div>
 
-        <ButtonBar errorMessage={status && status.submitError} inProgress={isSubmitting}>
-          <ActionGroup className="pf-c-form">
-            <Button
-              type="submit"
-              variant={ButtonVariant.primary}
-              isDisabled={!dirty || !_.isEmpty(errors) || !_.isEmpty(taskErrors)}
-              data-test-id="import-git-create-button"
-            >
-              {editingPipeline ? 'Save' : 'Create'}
-            </Button>
-            <Button type="button" variant={ButtonVariant.secondary} onClick={handleReset}>
-              Cancel
-            </Button>
-          </ActionGroup>
-        </ButtonBar>
-      </div>
-      <Sidebar open={!!selectedTask} onRequestClose={() => setSelectedTask(null)}>
-        {() => (
-          <TaskSidebar
-            resourceList={values.resources || []}
-            errorMap={taskErrors}
-            updateErrorMap={(newTaskErrors) => setTaskErrors(pruneErrors(newTaskErrors))}
-            setFieldValue={setFieldValue}
-            onRemoveTask={(taskName) => {
-              removeTaskModal(taskName, () => {
-                setSelectedTask(null);
-                setFieldValue(
-                  'tasks',
-                  values.tasks.filter((task) => task.name !== taskName),
-                );
-              });
-            }}
-            selectedPipelineTaskIndex={selectedTask.taskIndex}
-            taskResource={selectedTask.resource}
-          />
-        )}
-      </Sidebar>
-    </Form>
+          <ButtonBar inProgress={isSubmitting}>
+            <ActionGroup className="pf-c-form">
+              <Button
+                type="submit"
+                variant={ButtonVariant.primary}
+                isDisabled={
+                  !dirty || !_.isEmpty(errors) || !_.isEmpty(status) || values.tasks.length === 0
+                }
+                data-test-id="import-git-create-button"
+              >
+                {existingPipeline ? 'Save' : 'Create'}
+              </Button>
+              <Button type="button" variant={ButtonVariant.secondary} onClick={handleReset}>
+                Cancel
+              </Button>
+            </ActionGroup>
+          </ButtonBar>
+          <Sidebar open={!!selectedTask} onRequestClose={() => setSelectedTask(null)}>
+            {() => (
+              <TaskSidebar
+                resourceList={values.resources || []}
+                errorMap={(status as TaskErrorMap) || {}}
```

Comment:
Why the cast?

Reply:
Hold over from the previous design I was working with... there was a type error I couldn't figure out, so I casted it to get around it.